packages/sdk/app-framework/src/plugin-intent/intent-dispatcher.ts

```diff
@@ -206,36 +207,38 @@ export const createDispatcher = (
           }
         }
         if (result.error) {
-          break;
+          yield* Effect.fail(result.error);
         }
       }
 
       const results = yield* resultsRef.get;
       const result = results[0];
-      if (result) {
-        yield* Ref.update(historyRef, (history) => {
-          const next = [...history, results];
-          if (next.length > historyLimit) {
-            next.splice(0, next.length - historyLimit);
-          }
-          return next;
-        });
-
-        if (result.undoable && isUndoable(results)) {
-          // TODO(wittjosiah): Is there a better way to handle showing undo for chains?
-          yield* dispatch(createIntent(IntentAction.ShowUndo, { message: result.undoable.message }));
+      yield* Ref.update(historyRef, (history) => {
+        const next = [...history, results];
+        if (next.length > historyLimit) {
+          next.splice(0, next.length - historyLimit);
         }
-
-        return pick(result, ['data', 'error']);
-      } else {
-        return { data: {}, error: new Error('No results') };
+        return next;
+      });
+
+      if (result.undoable && isUndoable(results)) {
+        // TODO(wittjosiah): Is there a better way to handle showing undo for chains?
+        yield* pipe(
+          dispatch(createIntent(IntentAction.ShowUndo, { message: result.undoable.message })),
+          Effect.catchSome((err) =>
+            err instanceof NoResolversError ? Option.some(Effect.succeed(undefined)) : Option.none(),
+          ),
+        );
```

Comment:
Do we need to log here?

Reply:
I wondered about that but decided it wasn't worth it because there'd be no way to suppress it if you actually didn't want to provide a handler